src/util/system.cpp

```diff
@@ -443,6 +443,15 @@ const fs::path& ArgsManager::GetDataDir(bool net_specific) const
     } else {
         path = GetDefaultDataDir();
     }
+
+    if (fs::is_symlink(path)) {
+        path = fs::read_symlink(path);
+        if (!fs::is_directory(path)) {
+            path = "";
+            return path;
+        }
```

Comment:
What scenarios these lines are required in?

Reply:
A symbolic link may exist but point to a non existing location. For example, if your symbolic link point to an external hard drive, but it is not mounted.

Reply:
For example ![image](https://user-images.githubusercontent.com/9780671/152635687-e546aff8-e75d-4033-b94a-3c76d4830156.png)